security-admin/src/main/java/org/apache/ranger/security/web/filter/AltiscaleRangerAuthFilter.java

```diff
@@ -0,0 +1,477 @@
+										if(ustr != -1){
+											int andStr = s.indexOf("&", ustr);
+											if(andStr != -1){
+												try{
+													userName = s.substring(ustr+2, andStr);
+												}catch(Exception e){
+													userName = null;
+												}
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		// if security context does not have a user session, authenticate the security context and create a session
+		if(!isAuthenticated()){
+			String defaultUserRole = "ROLE_USER";
+			//if the userName is found on the token then log into ranger using the same user
+			if (userName != null && !userName.trim().isEmpty()) {
+				final List<GrantedAuthority> grantedAuths = new ArrayList<>();
+				grantedAuths.add(new SimpleGrantedAuthority(defaultUserRole));
+				final UserDetails principal = new User(userName, "",grantedAuths);
+				final Authentication finalAuthentication = new UsernamePasswordAuthenticationToken(principal, "", grantedAuths);
+				WebAuthenticationDetails webDetails = new WebAuthenticationDetails(request);
+				((AbstractAuthenticationToken) finalAuthentication).setDetails(webDetails);
+				RangerAuthenticationProvider authenticationProvider = new RangerAuthenticationProvider();
+				authenticationProvider.setAlt_ssoEnabled(true);
+				Authentication authentication = authenticationProvider.authenticate(finalAuthentication);
+				authentication = getGrantedAuthority(authentication);
+				SecurityContextHolder.getContext().setAuthentication(authentication);
+			}
+		}
+		// Delegate call to next filters
+		super.doFilter(filterChain, request, response);
+	}
+
+	private Authentication getGrantedAuthority(Authentication authentication) {
+		UsernamePasswordAuthenticationToken result=null;
+		if(authentication!=null && authentication.isAuthenticated()){
+			final List<GrantedAuthority> grantedAuths=getAuthorities(authentication.getName().toString());
+			final UserDetails userDetails = new User(authentication.getName().toString(), authentication.getCredentials().toString(),grantedAuths);
+			result = new UsernamePasswordAuthenticationToken(userDetails,authentication.getCredentials(),grantedAuths);
+			result.setDetails(authentication.getDetails());
+			return result;
+		}
+		return authentication;
+	}
+
+	private List<GrantedAuthority> getAuthorities(String username) {
+		Collection<String> roleList=userMgr.getRolesByLoginId(username);
+		final List<GrantedAuthority> grantedAuths = new ArrayList<>();
+		for(String role:roleList){
+			grantedAuths.add(new SimpleGrantedAuthority(role));
+		}
+		return grantedAuths;
+	}
+
+	private boolean isAuthenticated() {
+		Authentication existingAuth = SecurityContextHolder.getContext().getAuthentication();
+		return (existingAuth != null && existingAuth.isAuthenticated());
+	}
+
+	protected static ServletContext noContext = new ServletContext() {
+
+		@Override
+		public void setSessionTrackingModes(
+				Set<SessionTrackingMode> sessionTrackingModes) {
```

Comment:
indentation?

Reply:
I checked this and it is a trailing line in the parameter.